src/input_common/gcadapter/gc_adapter.cpp

```diff
@@ -193,10 +195,10 @@ void Adapter::UpdateVibrations() {
     // Modulate vibrations to 8 states
     vibration_counter = (vibration_counter + 1) % vibration_states;
 
-    for (size_t i = 0; i < pads.size(); ++i) {
-        const bool vibrate = pads[i].rumble_amplitude > vibration_counter;
-        vibration_changed |= vibrate != pads[i].enable_vibration;
-        pads[i].enable_vibration = vibrate;
+    for (GCController pad : pads) {
```

Comment:
```suggestion
    for (GCController& pad : pads) {
```

Otherwise this is a bug.